src/lib.rs

```diff
@@ -125,6 +127,40 @@ pub struct Event {
     pub writable: bool,
 }
 
+/// The mode in which the poller waits for I/O events.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[non_exhaustive]
+pub enum PollMode {
+    /// Poll in oneshot mode.
+    ///
+    /// In this mode, the poller will only deliver one event per file descriptor or socket.
+    /// Once an event has been delivered, interest in the event needs to be re-enabled
+    /// by calling `Poller::modify` or `Poller::add`.
+    ///
+    /// This is the default mode.
+    Oneshot,
+
+    /// Poll in level-triggered mode.
+    ///
+    /// Once an event has been delivered, polling will continue to deliver that event
+    /// until interest in the event is disabled by calling `Poller::modify` or `Poller::delete`.
+    ///
+    /// Not all operating system support this mode. Trying to register a file descriptor with
+    /// this mode in an unsupported operating system will raise an error. You can check if
+    /// the operating system supports this mode by calling `Poller::supports_level`.
+    Level,
+
+    /// Poll in edge-triggered mode.
+    ///
+    /// Once an event has been delivered, polling will not deliver that event again unless
+    /// a new event occurs.
+    ///
+    /// Not all operating system support this mode. Trying to register a file descriptor with
+    /// this mode in an unsupported operating system will raise an error. You can check if
+    /// the operating system supports this mode by calling `Poller::supports_edge`.
+    Edge,
```

Comment:
wouldn't it make more sense to only provide these enum values if the platform supports them, instead of some weird fallback? And transform the `supports_level` and `supports_edge` "methods" into constants? (idk what an effective way to handle that in downstream build scripts or cfg stmts would be, or if downstream users simply could copy "our" `cfg` restrictions...

Reply:
> wouldn't it make more sense to only provide these enum values if the platform supports them, instead of some weird fallback? 

It doesn't fall back, it errors out. This prevents cross-platform code from raising a compile-time error if, e.g, a certain code branch uses edge triggered mode.

The goal of this PR is to allow `polling` to be used in `calloop`, where edge triggered support already exists.

> And transform the `supports_level` and `supports_edge` "methods" into constants? 

The reason why I used methods instead of constants is so that is portable to potential future systems where edge/level triggered support is dependent on variable factors (e.g. kernel version).